compiler/modules/types/cell.bal

```diff
@@ -128,10 +128,46 @@ function intersectCellAtomicType(CellAtomicType c1, CellAtomicType c2) returns C
     return { ty, mut };
 }
 
+function cellSubtypeUnion(SubtypeData t1, SubtypeData t2) returns SubtypeData {
+    SubtypeData subtypeData = bddSubtypeUnion(t1, t2);
+    return cellFixSubtypeData(subtypeData);
+}
+
+function cellSubtypeIntersect(SubtypeData t1, SubtypeData t2) returns SubtypeData {
+    SubtypeData subtypeData = bddSubtypeIntersect(t1, t2);
+    return cellFixSubtypeData(subtypeData);
+}
+
+function cellSubtypeDiff(SubtypeData t1, SubtypeData t2) returns SubtypeData {
+    SubtypeData subtypeData = bddSubtypeDiff(t1, t2);
+    return cellFixSubtypeData(subtypeData);
+}
+
+function cellSubtypeComplement(SubtypeData t) returns SubtypeData {
+    SubtypeData subtypeData = bddSubtypeComplement(t);
+    return cellFixSubtypeData(subtypeData);
+}
+
+// SubtypeData being a boolean would result in a BasicTypeBitSet which could either be 0 or 1 << BT_CELL.
+// This is to avoid cell type being a BasicTypeBitSet, as we don't want to lose the cell wrapper.
+function cellFixSubtypeData(SubtypeData subtypeData) returns SubtypeData {
```

Comment:
Well, when we have two cells in which the diff is empty, we end up creating `0` as the resultant semType. This is contrary to having MemberSemType enforced with the following type definition.
``` bal 
public type MemberSemType readonly & record {|
    0 all = 0;
    BasicTypeBitSet some = CELL;
    ProperSubtypeData[1] subtypeDataList;
|};
```
Therefore I did this fix.

The option would be to make MemberSemType 
```bal
type MemberSemType NEVER | CELL | record {| 0 all = 0 BasicTypeBitSet some = CELL; ProperSubtypeData[1] subtypeDataList; |};
```
However, with the second option we will not have the cell wrapper all the time.